lib/pengine/bundle.c

```diff
@@ -1516,7 +1516,8 @@ pe__bundle_xml(pcmk__output_t *out, va_list args)
 
         CRM_ASSERT(replica);
 
-        if (!pe__rsc_running_on_any_node_in_list(replica->container, only_show)) {
+        if ((replica->container->fns->active(rsc, TRUE) && !pe__rsc_running_on_any_node_in_list(replica->container, only_show)) ||
+            (!replica->container->fns->active(rsc, FALSE) && !pe__rsc_allowed_on_any_node_in_list(replica->container, only_show))) {
```

Comment:
The active() argument has to be the same resource owning the function, e.g. 

    replica->container->fns->active(replica->container, TRUE)

which will make these even longer, so functionizing this test is probably a good idea (though I tremble to think of naming it).

Also, I think you want FALSE for both active() calls, i.e. whether the resource is at least partially active.

Reply:
There, that wasn't as bad as I was thinking.  The name wasn't difficult at all.